src/main/java/com/monitorjbl/xlsx/StreamingReader.java

```diff
@@ -111,11 +127,57 @@ private void handleEvent(XMLEvent event) throws SAXException {
         currentCell = new StreamingCell(CellReference.convertColStringToIndex(coord[0]), Integer.parseInt(coord[1]) - 1);
 
         Attribute type = startElement.getAttributeByName(new QName("t"));
+/*ForgetMeNot--ibell--20150911--old code which didn't determine numeric vs date data types.
         if(type != null) {
           currentCell.setType(type.getValue());
         }
+*/
+        //
+        // This block will determine the value type and set the appropriate
+        // XSSF data type (used to format numeric cell values).
+        //
+        final String typeValue = (type != null) ? type.getValue() : "";
+        if ("b".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.BOOLEAN);
+        } else if ("e".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.ERROR);
+        } else if ("str".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.FORMULA);
+        } else if ("inlineStr".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.INLINE_STRING);
+        } else if ("s".equals(typeValue)) {
+          currentCell.setXssfDataType(StreamingCell.XSSF_DATA_TYPE.SST_INDEX);
+        } else {
```

Comment:
I'm guessing there's a type string for numerics that you're omitting for brevity here. I'd prefer that you explicitly check for it and have an `else` block that throws an exception if reached.